docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+  +----------+----------+-----------+-----------+----------+---------+-------------+--------+
+
+.. _mapgen-placement:
+
+Player Placement
+----------------
+
+.. table:: Mode chosen by the generator to generate start positions
+  :widths: auto
+  :align: right
+
+  ============ =======
+  Generator    Default
+  ============ =======
+  ``FAIR``     Placed by the generator
+  ``FRACTURE`` ``ALL``
+  ``FRACTAL``  ``ALL``
+  ``ISLAND``   Always ``SINGLE``
+  ``RANDOM``   ``2or3``
+  Scenarios    ``ALL``
+  ============ =======
+
+The final step in the map generator is to to place players on the map. The
+method used to do so is set by the user, but it can also can depend on the
+generator if ``startpos`` is set to ``DEFAULT``, as listed on the right.
+Starting positions are allocated using the chosen method. If a method fails,
+another method is tried in the following order: ``SINGLE``, ``2or3``, ``ALL``,
+and finally ``VARIABLE``.
+
+Placement tries to find fair starting positions using a "tile value" metric,
+computed as the sum of all outputs produced by the tile (food, production, and
+trade). If the initial workers can build a road, irrigation, or mine, half of
+the best possible bonus is counted towards the value of the tile, rounded down.
+Specials are taken into account but no government bonus is applied. This gives
+the following values for common terrains:
+
+.. table:: Tile value for the most common terrains
+  :widths: auto
+  :align: center
+
+  ============ =============== ====================== =====
+  Terrain      Food/Prod/Trade Road + Irrigation/Mine Value
+  ============ =============== ====================== =====
+  Forest       1/2/0           \-                     3
+  Grassland    2/0/0           1 + 1/-                3
+  Ocean        1/0/2           \-                     3
+  Desert       0/1/0           1 + 1/1                2
+  Plains       1/0/0           1 + 1/-                2
+  Hills        1/0/0           \- + 1/3               2
+  Jungle       1/0/0           \-                     1
+  Mountains    1/0/0           \- + -/1               1
+  Swamp        1/0/0           \-                     1
+  Tundra       1/0/0           \- + 1/-               1
+  ============ =============== ====================== =====
+
+The initial city radius is then taken into account. This is done by setting the
+tile value to zero if, within the city radius, more tiles are worse than better.
+(So the value of a wheat tile surrounded by grass and a pheasant is set to
+zero because all tiles except the pheasant are strictly worse.)
+After this step, the map of tile values is smoothed out using a Gaussian filter
+of width 1 and the value of ocean tiles is set to zero as they cannot be used as
+starting positions. The total value of every island is computed by summing over
+all tiles. Finally, tile values are normalized to the range [0, 1000).
```

Comment:
you have a [ and a ). do you want to use an open ( instead of a [?

Reply:
This is one of the common mathematical notations for an open-ended interval, along with [0, 1000[. I could also change to 0-999 but I prefer to keep 1000 since it's easier to remember.